api/app/Http/Controllers/Admin/AdminController.php

```diff
@@ -84,6 +89,54 @@ public function fetchUser($identifier)
         ]);
     }
 
+    public function blockUser(Request $request, UserActionService $userActionService)
+    {
+        $request->validate([
+            'user_id' => 'required',
+            'reason' => 'required|string|max:1000',
+        ]);
```

Comment:
let's create dedicated request to clean this controlelr method (same for unblock - Maybe can re-use it?)